### PR TITLE
fix: Unable to create a conversation with a service SQSERVICES-1394

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1394" title="SQSERVICES-1394" target="_blank">BugSQSERVICES-1394</a>  [iOS] Unable to start a 1:1 conversation with a service
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

if we want to create a conversation with bot. We need to set the `conversation.allowServices` to true.  So `service` would be added to the accessRoles and the payload is going to be correct. The reason wasn't working was because we didn't add "service" to the accessRoles so the payload was incorrect.

The error we got from backend was. `403 "invalid-conversation"` That response appears when adding or removing services. Also that meant that maybe the set of accessRoles was incorrect when trying to create a conversation with a service. 

So by setting `conversation.allowServices = true` we're fixing the issue. We know that accessRoles and the payload is going to be updated correctly.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by KaterinaWire — APPROVED** [review]

File: Source/Data Model/ServiceUser.swift [hunk 1]
```diff
@@ -209,6 +209,10 @@ public extension ServiceUser {
         conversation.creator = selfUser
         conversation.team = selfUser.team
 
+        if conversation.allowServices == false {
+            conversation.allowServices = true
+        }
```
Comment (johnxnguyen) [inline Source/Data Model/ServiceUser.swift:214]: I don't think the `if` is necessary here, you can just set it to `true`.

**Review by johnxnguyen — COMMENTED** [review]
I thought that when creating a conversation, the default value for the access roles already includes services. Is that not true?

File: Source/Data Model/ServiceUser.swift (hunk 1 above)
Comment (agisilaos) [reply Source/Data Model/ServiceUser.swift]: Done!

**Comment by agisilaos** [comment]
> I thought that when creating a conversation, the default value for the access roles already includes services. Is that not true?

When creating a (group) conversation allow Guests and Services toggle is turned on by default. But user Can turn them off before creating that conversation. We didn't take into account creating a convo with a service though and the default value for accessRoles wasn't including service.

**Review by johnxnguyen — APPROVED** [review]

**Comment by agisilaos** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @agisilaos! Merging PR for command **release patch** for PR #**1700**. Will create a **patch** release after merged.
Next version is 416.0.1.

**Pull request merged**